interface/src/app/experiment/experiment.component.js

```diff
@@ -0,0 +1,69 @@
+angular.module('adage.experiment', [
+  'adage.experiment.service',
+  'adage.analyze.sample',   // TODO fix after Sample is also refactored
+  'statusBar'
+])
+
+.component('experimentDetail', {
+  templateUrl: 'experiment/experiment.component.tpl.html',
+  bindings: {
+    id: '<'
+  },
+  controller: [
+    'Experiment', 'Sample', '$log',
+    function(Experiment, Sample, $log) {
+      var ctrl = this;
+
+      ctrl.$onChanges = function(changesObj) {
+        ctrl.show(changesObj.id.currentValue);
+      };
+      ctrl.makeHref = Experiment.makeHref;
+      ctrl.experiment = {
+        status: '',
+        relatedSamples: []
+      };
+
+      var queryError = function(responseObject, responseHeaders) {
+        $log.warn('Query errored with: ' + responseObject);
+        ctrl.experiment.status = 'Query failed.';
+      };
+
+      ctrl.show = function(id) {
+        if (!id) {
+          $log.warn('experiment.show called with id', id);
```

Comment:
Do you think the warning message can be more specific? The word `experiment` in `experiment.show` sounds very generic.

Reply:
I thought about this a little. I want the message to be somewhat generic since this is a reusable component now. On the other hand, I think we can do better by using the name of this module instead, so it could say this instead:

```
'adage.experiment: show() called with id'
```

Reply:
Yes, I think the new message is much better.